### PR TITLE
Synthetics Bugs

### PR DESCRIPTION
**Pull request by hyperiongeo**
Matt

I had some time so I took a look at this.  Turns out the problem was with the wavelet.  At some point in the past, the input into the wavelet calculation was made in seconds rather than milliseconds but the ricker calculation expected ms.  I changed the ricker calculation to accept seconds (which I believe matches the bruges version).

During testing, I noticed a bug on the as_curve function so I fixed that as well with an improvement over the handling of the log output interval.  I also made a change in the make_synthetic where the user can add their chosen sonic and density mneonics rather than having to rename them to DT and RHOB (which remain the defaults).

Hope this meets with your approval.

Corey

### CODE REVIEW

**Comment by hyperiongeo** [comment]
I've reviewed this in light of the newer updates and the synthetic issues remain.  This should merge into the existing code without any issues I could identify.

File: welly/synthetic.py [hunk 1]
```diff
@@ -55,14 +55,19 @@ def basis(self):
...
         return np.arange(self.start, self.stop - precision_adj, self.dt)
 
-    def as_curve(self, data):
+    def as_curve(self, start=None, stop=None, step=0.1524):
         """
         Get the synthetic as a Curve, in depth. Facilitates plotting along-
         side other curve data.
         """
-        params = {'mnemonic': 'SYN'}
+        params = {'start': start or getattr(self, 'depth_start', None),
```
Comment (frank1010111) [inline welly/synthetic.py:63]: this will behave unexpectedly if start=0

File: welly/synthetic.py [hunk 2]
```diff
@@ -55,14 +55,19 @@ def basis(self):
...
+                  'mnemonic': 'SYN',
+                  'step': step
+                  }
 
-        return Curve(data, **params)
+        stop = stop or getattr(self, 'depth_stop', None)
```
Comment (frank1010111) [inline welly/synthetic.py:68]: this will behave unexpectedly if (albeit this is unlikely) stop = 0

**Review by frank1010111 — COMMENTED** [review]
this looks great. I've just got two nit-picks about testing None for truthiness and the obligatory question "do you have unit tests?"

**Comment by hyperiongeo** [comment]
Frank - I've reviewed this based on your comments and updated the as_curve function to eliminate the Nones.  Added a few conditionals to account for depths that may not be realistic.  Also added a unit test for the as_curve.

**Pull request merged**